src/App.tsx

```diff
@@ -1,18 +1,34 @@
-import { Navigation } from "./components/layout/Navigation";
+import { RequireAuth } from "components/auth/RequireAuth";
 import React, { FunctionComponent } from "react";
-import { HashRouter as Router, Route, Switch } from "react-router-dom";
+import {
+  HashRouter as Router,
+  Route,
+  Switch,
+  useParams,
+} from "react-router-dom";
 import "./App.scss";
-import { BeaconsGateway } from "./gateways/BeaconsGateway";
-import { Footer } from "./components/layout/Footer";
 import { AuthWrapper } from "./components/auth/AuthWrapper";
-import { RequireAuth } from "components/auth/RequireAuth";
-import { BeaconRecords } from "./components/BeaconRecords";
-import { Beacon } from "./components/Beacon";
 import { Home } from "./components/Home";
+import { Footer } from "./components/layout/Footer";
+import { Navigation } from "./components/layout/Navigation";
+import { BeaconsGateway } from "./gateways/BeaconsGateway";
+import { BeaconRecordsListView } from "./views/BeaconRecordsListView";
+import { SingleBeaconRecordView } from "./views/SingleBeaconRecordView";
+
+interface ResourceParams {
+  id: string;
+}
 
 const App: FunctionComponent = () => {
   const beaconsGateway = new BeaconsGateway();
 
+  const SingleBeaconRecordViewWithParam: FunctionComponent = () => {
+    const { id } = useParams<ResourceParams>();
+    return (
+      <SingleBeaconRecordView beaconsGateway={beaconsGateway} beaconId={id} />
+    );
+  };
```

Comment:
See https://reactrouter.com/web/example/url-params.

`Resource` is a general term to apply to any view we would want to access by parameterised URL, in this case a `beacon` but also `owner`, `use` etc.

`<SingleBeaconRecordViewWithParam>` wraps the `<SingleBeaconRecordView>` component and injects the `id` parameter to `beaconId` so that `<SingleBeaconRecordView>` doesn't have knowledge of React Router and the URL parameter.  Hope that makes sense!  Might be simpler to just call `useParams` from within the `SingleBeaconRecordView`?